# Data Exploration

In [1]:
import pandas as pd

In [2]:
DATA_FOLDER = "../data/"

## Cell line data

In [3]:
cell_lines = pd.read_csv(DATA_FOLDER + "Cell_Lines_Details.csv")
print(cell_lines.shape)
cell_lines.head()

(1001, 13)


,Sample Name,COSMIC identifier,Whole Exome Sequencing (WES),Copy Number Alterations (CNA),Gene Expression,Methylation,Drug\rResponse,GDSC\rTissue descriptor 1,GDSC\rTissue\rdescriptor 2,Cancer Type\r(matching TCGA label),Microsatellite \rinstability Status (MSI),Screen Medium,Growth Properties
0,A253,906794,Y,Y,Y,Y,Y,aero_dig_tract,head and neck,NaN,MSS/MSI-L,D/F12,Adherent
1,BB30-HNC,753531,Y,Y,Y,Y,Y,aero_dig_tract,head and neck,HNSC,MSS/MSI-L,D/F12,Adherent
2,BB49-HNC,753532,Y,Y,Y,Y,Y,aero_dig_tract,head and neck,HNSC,MSS/MSI-L,D/F12,Adherent
3,BHY,753535,Y,Y,Y,Y,Y,aero_dig_tract,head and neck,HNSC,MSS/MSI-L,D/F12,Adherent
4,BICR10,1290724,Y,Y,Y,Y,Y,aero_dig_tract,head and neck,HNSC,MSS/MSI-L,D/F12,Adherent


In [4]:
list(cell_lines.columns.values)

['Sample Name',
 'COSMIC identifier',
 'Whole Exome Sequencing (WES)',
 'Copy Number Alterations (CNA)',
 'Gene Expression',
 'Methylation',
 'Drug\rResponse',
 'GDSC\rTissue descriptor 1',
 'GDSC\rTissue\rdescriptor 2',
 'Cancer Type\r(matching TCGA label)',
 'Microsatellite \rinstability Status (MSI)',
 'Screen Medium',
 'Growth Properties']

In [7]:

cell_features = pd.read_csv(DATA_FOLDER+"Cell_Line_Features_PANCAN_simple_MOBEM.tsv", sep='\t')
print(cell_features.shape)
cell_features.head()

(1073, 1002)


,Unnamed: 0,1287381,924100,910924,687561,1287706,687452,906798,906797,906800,...,910545,1524416,971774,930083,909700,909701,753620,905965,1299061,1331040
0,ABCB1_mut,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
1,ABL2_mut,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
2,ACACA_mut,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3,ACVR1B_mut,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
4,ACVR2A_mut,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


Each cell line has 1002 features. The current TabPFN version is built for 50k data points and 2000 features. This leads to 2000 - 1002 = 998 for drug features 